poetry/installation/pip_installer.py

```diff
@@ -210,27 +210,27 @@ def install_directory(self, package):
 
         setup = os.path.join(req, "setup.py")
         has_setup = os.path.exists(setup)
-        if has_poetry and package.develop and not package.build_script:
-            # This is a Poetry package in editable mode
-            # we can use the EditableBuilder without going through pip
-            # to install it, unless it has a build script.
-            builder = EditableBuilder(
-                Factory().create_poetry(pyproject.parent), self._env, NullIO()
-            )
-            builder.build()
+        if has_poetry:
+            package_poetry = Factory().create_poetry(pyproject.parent)
```

Comment:
We should probably only instantiate if it is requred.

Reply:
We need it just below for the `build_script` check. Or do you have something else in mind?

Reply:
I was more thinking of using an `isinstance` or `hasattr` check. But this is not strictly required.